src/hooks/useFetch.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+const fetchUrl = (url, after, count, limit = 100, time = 'year') => {
+  const urlParams = `t=${time}&limit=${limit}${
+    count && after ? `&after=${after}&count${count}` : ''
+  }`;
+  return `${url}/top.json?${urlParams}`;
+};
+
+const makeRequest = async (url) => {
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    throw new Error('Unable to fetch data');
+  }
+
+  const data = await res.json();
+  return data;
+};
+
+const fetchPosts = async (
+  url,
+  posts = [],
+  count = null,
+  after = null,
+  numRequests = 500,
+) => {
+  if (count >= numRequests) {
+    return posts;
+  }
+
+  const currentUrl = fetchUrl(url, after, count);
+
+  const data = await makeRequest(currentUrl);
+
+  const recordsRetrieved = data.data.dist;
+  posts.push(...data.data.children);
+
+  if (recordsRetrieved < 100) {
+    return posts;
+  }
+
+  const currentCount = count + recordsRetrieved;
+  const currentAfter = data.data.after;
+
+  return fetchPosts(url, posts, currentCount, currentAfter);
+};
+
+const useFetch = (url) => {
```

Comment:
Ah OK, I guess I understand now thy you pass the url as parameter. Was the idea to have a more or less generic fetch hook?

Since this is very specific to fetching reddit posts I'd probably rename it to `useFetchPosts` and take the subreddit as parameter

Reply:
The issue was passing subreddit and using it as the useEffect dependency which was causing problems as I typed into the input.

Reply:
When I change the dependency to subreddit and start typing in the box, I get this in the console:

![image](https://user-images.githubusercontent.com/55104465/84159451-13c9b500-aa3b-11ea-9268-26721ad48952.png)

Then I am unable to get any requests to work at all unless I refresh the page completely.

Reply:
or I get this:

![image](https://user-images.githubusercontent.com/55104465/84161236-0f060080-aa3d-11ea-84a1-0315b8326962.png)

Reply:
True, it shouldn't use the form input value. I left a comment with another suggestion above